subprojects/core/src/main/java/org/gradle/initialization/DefaultCommandLineConverter.java

```diff
@@ -203,11 +198,11 @@ public StartParameter convert(final ParsedCommandLine options, final StartParame
             startParameter.setNoBuildScan(true);
         }
 
-        if (options.hasOption(BUILD_CACHE)) {
+        if (options.hasOption(BUILD_CACHE.getCommandLineOption().getEnabledOption())) {
```

Comment:
I think a part of the goal here is to have CLI consistency, which means that for _all_ boolean options we can have the same logic. To me, that means the ability to loop over some filtered collection of `GradleBuildOption`s in one place and apply logic there. 

That also means we would introduce `--no-parallel` and `--no-configure-on-demand` as well. We should have these IMO.

Reply:
I had the same comment above:

> Currently we cannot just iterate over all GradleBuildOption as the creation of these options for the command line parser isn't done in one single place.

It would mean plenty of additional work though as we create options in different places right now.